src/stacks-hierarchy/Root_ChooseTicketReceiverScreen/Root_ChooseTicketReceiverScreen.tsx

```diff
@@ -23,6 +24,21 @@ export const Root_ChooseTicketReceiverScreen: React.FC<Props> = ({
 
   const focusRef = useFocusOnLoad();
 
+  const handleValidatePhoneNumber = async (phoneNumber: string) => {
+    const result = await getCustomerAccountId(phoneNumber);
+    if (result) {
+      switch (typeof result) {
+        case 'string': // customer account id successfully returned
+          return result;
+        case 'object': // phone number not associated to any account or invalid number
+          return 'no_associated_account';
+        case 'undefined': // unknown error
+          return 'unknown_error';
```

Comment:
For this case, I think the result is not undefined never because of the condition.

Reply:
What about:

return typeof result === 'string' ? result : 'no_associated_account';

Reply:
the `getCustomerAccountId` function may return 3 types: string (if request success), object (if request fails and there's a message from backend), and undefined (other errors, server issue, for instance), so I want these types to be supported, what do you think?

Reply:
I mean if the 3 states are needed then is OK but because there is a condition first that check result is not undefined but inside ask for undefined type which was excluded at the top then I just see two cases if response a string then is good if not then an error that includes undefined and object.